pkgs/test/test/runner/engine_test.dart

```diff
@@ -58,6 +59,19 @@ void main() {
     engine.suiteSink.close();
   });
 
+  test("returns failure if all tests do not complete", () async {
+    var engine = declareEngine(() {
+      for (var i = 0; i < 3; i++) {
+        test("test ${i + 1}", () async {
+          await Future.delayed(Duration(seconds: 5));
```

Comment:
can we use a `Completer`  that we never `.complete()` to make this more explicit than a delay?

Reply:
Discussed offline. These tests are odd so I can't open non-completed tasks. We came up with an alternative that I'll implement though.